lib/interfaces.ts

```diff
@@ -0,0 +1,66 @@
+import { ComputedRef, DeepReadonly, Ref, UnwrapNestedRefs, WatchCallback } from 'vue'
+
+export interface IModule<
+  S extends IState,
+  G extends IGetters,
+  M extends IMutations,
+  SM extends ISubModules = never,
+> {
+  readonly options: IModuleOptions<S, G, M, SM>
+  flatten(): IFlattenedModule<S, G, M, SM>
+}
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ISubModules = Record<string, IModule<any, any, any, any>>
+
+export type IState = Record<string, any>
+
+export type IGetters = Record<string, () => any>
+
+export type IMutations = Record<string, (...args: any) => void>
+
+export interface IPlugin<S extends IState> {
+  onStateInit?: (state: S) => S
+  onDataChange?: WatchCallback<UnwrapNestedRefs<S>, UnwrapNestedRefs<S>>
+}
+/* eslint-enable */
+
+export interface IModuleOptions<
```

Comment:
I modified the types so that the plugin type is not a generic parameter. I think parameterizing below as `IPlugin<S>[]` is sufficient, though maybe I've missed something.